src/units/unit.cpp

```diff
@@ -2428,6 +2428,33 @@ void unit::apply_builtin_effect(std::string apply_to, const config& effect)
 			level_ += lexical_cast_default<int>(increase);
 		}
 	}
+
+	// In case the effect carries EventWML, apply it now
+	if(resources::game_events && resources::lua_kernel) {
+		config events;
+		for(const config& effect_event : effect.child_range("event")) {
+			events.add_child("event", effect_event);
+		}
+		for(const config& abilities : effect.child_range("abilities")) {
+			for(const config::any_child child : abilities.all_children_range()) {
+				const config& ability = child.cfg;
+				for(const config& ability_event : ability.child_range("event")) {
+					events.add_child("event", ability_event);
+				}
+			}
+		}
+		for(const config& attack : effect.child_range("attack")) {
```

Comment:
Can we limit this to `if(effect["apply_to"] == "new_attack")`?

Also, you're missing the case where `effect["apply_to"] == "attack"`. In that case, you have a `[specials]` tag directly in `[effect]` without an intervening `[attack]` tag.